app/controllers/api/v1/messages_controller.rb

```diff
@@ -58,8 +62,38 @@ def mark_read
         render json: @message, serializer: serializer
       end
 
+      api :PUT, "/v1/messages/mark_all_read", "Mark all messages as read"
+      def mark_all_read
+        @subscriptions = Subscription.unread.of_active_user
+        if params[:scope].present?
+          @subscriptions = apply_scope(@subscriptions.joins(:message), params[:scope])
+        end
+        @subscriptions.update_all state: 'read'
+        render json: {}
+      end
+
       private
 
+      def apply_scope(records, scope)
+        return records unless ALLOWED_SCOPES.include? scope
+        records.where("messages.#{scope}_id IS NOT NULL")
```

Comment:
This might get flagged up in auto scanning tools for SQL injection but I can see you've limited scope options.